crud/src/main/java/com/redhat/lightblue/assoc/ep/ExecutionPlan.java

```diff
@@ -225,11 +225,13 @@ public ExecutionPlan(QueryExpression requestQuery,
                         if (requestSort != null) {
                             last = new Source<>(new SortResults(block, last, requestSort));
                         }
-                        if (from != null) {
-                            last = new Source<>(new Skip(block, from.intValue(), last));
-                        }
-                        if (to != null) {
-                            last = new Source<>(new Limit(block, to.intValue() - from.intValue() + 1, last));
+                        if(!needsFinalFiltering) {
```

Comment:
Is there a reason to not just apply the limit after the filter step to eliminate duplication of logic?